macros/get_relation_from_string.sql

```diff
@@ -0,0 +1,47 @@
+{#
+Copyright (c) 2024-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+/*  Macro to allow for hardcoding of the source table reference in var('snowplow__conversion_path_source') 
+and var('snowplow__conversions_source'). */
+
+{% macro get_relation_from_string(full_table_reference) %}
+
+  {% set full_table_reference_without_quotes = full_table_reference | replace("`", "") | replace('"', '') %}
+  {% set parts = full_table_reference_without_quotes.split('.') %}
+  
+  {% if parts | length == 3 %}
+    {% set database = parts[0] %}
+    {% set schema = parts[1] %}
+    {% set table = parts[2] %}
+  
+  {% elif parts | length == 2 %}
+    {% set database = target.database %}
+    {% set schema = parts[0] %}
+    {% set table = parts[1] %}
+  
+  {% elif parts | length == 1 %}
+    {% set database = target.database %}
+    {% set schema = target.schema %}
+    {% set table = parts[0] %}
```

Comment:
Just wondering, should we take it from `this` instead of `target`? Because usually they would be in the derived schema which wouldn't be specifically the target?

Reply:
Yeah, good point